cli/src/config.rs

```diff
@@ -624,6 +625,34 @@ pub fn default_config_migrations() -> Vec<ConfigMigrationRule> {
             "core.watchman.register_snapshot_trigger",
             "core.watchman.register-snapshot-trigger",
         ),
+        // TODO: Delete in jj 0.34+
+        ConfigMigrationRule::custom(
+            |layer| {
+                if let Ok(Some(value)) = layer.look_up_item("git.sign-on-push") {
+                    value.as_bool().is_some()
+                } else {
+                    false
+                }
+            },
+            |layer| match layer.look_up_item("git.sign-on-push") {
+                Ok(Some(value)) => {
+                    let old_value = value
+                        .as_bool()
+                        .ok_or("git.sign-on-push expects a boolean")
+                        .map_err(|error| ConfigMigrateLayerError::Type {
+                            name: "git.sign-on-push".to_string(),
+                            error: error.into(),
```

Comment:
nit: `.ok_or_else(|| ..`

Perhaps, `.unwrap()` is also good because the type should have been tested.